anndata/tests/test_concatenate.py

```diff
@@ -406,6 +407,21 @@ def test_concatenate_obsm_outer(obsm_adatas, fill_val):
     pd.testing.assert_frame_equal(true_df, cur_df)
 
 
+@pytest.mark.parametrize(
+    ("axis_val", "shape"),
+    [
+        pytest.param("obs", (20, 10), id="obs"),
+        pytest.param(0, (20, 10), id="0"),
+        pytest.param("var", (10, 20), id="var"),
+        pytest.param(1, (10, 20), id="1"),
+    ],
+)
+def test_concat_axis_param(axis_val, shape):
+    a, b = gen_adata((10, 10)), gen_adata((10, 10))
+    c = concat([a, b], axis=axis_val)
+    assert c.shape == shape
```

Comment:
```suggestion
@pytest.mark.parametrize(
    ("axis", "axis_name"),
    [("obs", 0), ("var", 1)],
)
def test_concat_axis_param(axis, axis_name):
    a, b = gen_adata((10, 10)), gen_adata((10, 10))
    assert_equal(
        concat([a, b], axis=axis),
        concat([a, b], axis=axis_name)
    )
```

A bit more simple and more thorough I think.

I think this basically covers all changes that are really needed to the tests in this PR too, no?

Reply:
I think so too.